benchmarks/mx_formats/mm_bench.py

```diff
@@ -0,0 +1,200 @@
+        print(f"💾 Results saved to: {save_path}")
+
+
+def plot_tflops_comparison(df, save_path: Path):
+    plt.figure(figsize=(12, 6))
+    grouped = df.groupby(["K", "Format"])
+    k_values = sorted(df["K"].unique())
+    formats = df["Format"].unique()
+    m_value = df["M"].iloc[0]
+    n_value = df["N"].iloc[0]
+
+    # Plot MX kernel performance
+    for format in formats:
+        try:
+            tflops_values = [
+                grouped.get_group((k, format))["TFLOPS"].values[0] for k in k_values
+            ]
+            plt.plot(k_values, tflops_values, label=format)
+        except KeyError:
+            # Skip if this combination doesn't exist in the data
+            continue
+
+    plt.xlabel("K (Matrix Dimension)")
+    plt.ylabel("TFLOPS")
+
+    # Set y-axis to start at 0
+    plt.ylim(bottom=0)
+
+    title = f"MX Matrix Multiplication Performance \nM={m_value}, N={n_value}"
+    plt.title(title)
+
+    plt.legend()
+    plt.grid(True, which="both", ls="-", alpha=0.2)
+    plt.xticks(k_values, rotation=45, ha="right")
+    plt.tight_layout()
+
+    # Generate the file name and save in the same directory as the CSV file
+    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+    file_name = f"mx_{m_value}_{n_value}_{timestamp}.png"
+    graph_path = save_path.parent / file_name
+    plt.savefig(graph_path, dpi=300)
+    print(f"TFLOPS comparison plot saved as {graph_path}")
+
+
+def get_configs_varying_k(M: int = 8192, N: int = 8192) -> list[ExperimentConfig]:
+    shapes = [(M, K, N) for K in range(1024, 16385, 1024)]
+    formats = [Format.MX_FP8, Format.MX_FP4]
+
+    configs = [
+        ExperimentConfig(M=M, K=K, N=N, format=format)
+        for (M, K, N), format in itertools.product(shapes, formats)
+    ]
+    return configs
+
+
+def main(
+    save_path: str | None = None, M: int = 8192, N: int = 8192, graph: bool = False
+):
+    """Benchmark MX MatMul with different configurations and optionally graph results.
+
+    Args:
+        save_path (Optional[str], optional): Path to save the results. Defaults to None.
+        M (int, optional): Number of rows in the first matrix. Defaults to 8192.
+        N (int, optional): Number of columns in the second matrix. Defaults to 8192.
+        graph (bool, optional): Whether to create a graph of the results. Defaults to False.
+    """
+    torch.random.manual_seed(123)
+    configs = get_configs_varying_k(M, N)
+    results = []
+    if save_path is not None:
+        save_path = Path(save_path)
+        save_path = save_path.with_suffix(".csv")
+        save_path.parent.mkdir(parents=True, exist_ok=True)
+    for config in tqdm(configs):
+        result = run_experiment(config)
+        results.append(Experiment(config=config, result=result))
+    print_results(results, save_path)
+
+    if graph and save_path is not None:
+        df = pd.read_csv(save_path)
+        plot_tflops_comparison(df, save_path)
+
+
+if __name__ == "__main__":
```

Comment:
can we unify this code with https://github.com/pytorch/ao/blob/main/benchmarks/float8/bench_matmul.py instead?  I know the path says `float8` but it would be good to have it all in one place.

Reply:
I have added mxfp4 to that script. You can review it. Thank you

Reply:
thank you!